tools/test-proxy/typespec/main.tsp

```diff
@@ -0,0 +1,189 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/openapi3";
+
+using TypeSpec.Http;
+
+@service({
+    title: "Test-Proxy Record/Playback server"
+})
+@server("https://localhost:5001/", "The test-proxy default URL.")
+
+namespace TestProxy;
+
+model StartPayload {
+    "x-recording-file"?: string;
+    "x-recording-assets-file"?: string;
+}
+
+@route("/Record/Start")
+namespace RecordStart {
+    @post
+    op start(@body body: StartPayload): {
+        @statusCode statusCode: 200;
+        @header("x-recording-id") recordingId: string;
+    };
+}
+
+@route("/Record/Stop")
+namespace RecordStop {
+    model VariablePayload {
+        Variables: Record<string>;
```

Comment:
I need this to be `Dictionary<string, string>`. I know typescript supports `Record<T, V>`, so I'm a bit weirded out by this. It's a typespec specific issue.